drivers/adc-dac/ad74413r/ad74413r.c

```diff
@@ -1085,6 +1085,24 @@ int ad74413r_remove(struct ad74413r_desc *desc)
 	if (!desc)
 		return -EINVAL;
 
+	/* Perform a reset to bring the device in the default state */
+	if (desc->reset_gpio) {
```

Comment:
this looks like duplicated code. A separate function for hardware reset might fit. And then call it both in init function and remove function.